Tests/WorkspaceTests/PrebuiltsTests.swift

```diff
@@ -0,0 +1,545 @@
+                    ]
+                )
+            ],
+            dependencies: [
+                .sourceControl(
+                    url: "https://github.com/swiftlang/swift-syntax.git",
+                    requirement: .exact(try XCTUnwrap(Version(swiftSyntaxVersion)))
+                )
+            ]
+        )
+
+        let swiftSyntax = try MockPackage(
+            name: "swift-syntax",
+            url: "https://github.com/swiftlang/swift-syntax.git",
+            targets: [
+                MockTarget(name: "SwiftSyntaxMacrosTestSupport"),
+                MockTarget(name: "SwiftCompilerPlugin"),
+                MockTarget(name: "SwiftSyntaxMacros"),
+            ],
+            products: [
+                MockProduct(name: "SwiftSyntaxMacrosTestSupport", modules: ["SwiftSyntaxMacrosTestSupport"]),
+                MockProduct(name: "SwiftCompilerPlugin", modules: ["SwiftCompilerPlugin"]),
+                MockProduct(name: "SwiftSyntaxMacros", modules: ["SwiftSyntaxMacros"]),
+            ],
+            versions: ["600.0.1", "600.0.2"]
+        )
+
+        return (manifest, rootPackage, swiftSyntax)
+    }
+
+    func checkSettings(_ target: Module, usePrebuilt: Bool) throws {
+        if usePrebuilt {
+            let swiftFlags = try XCTUnwrap(target.buildSettings.assignments[.OTHER_SWIFT_FLAGS]).flatMap({ $0.values })
+            XCTAssertTrue(swiftFlags.contains("-I/tmp/ws/.build/prebuilts/swift-syntax/\(self.swiftVersion)-MacroSupport-macos_aarch64/Modules".fixwin))
+            XCTAssertTrue(swiftFlags.contains("-I/tmp/ws/.build/prebuilts/swift-syntax/\(self.swiftVersion)-MacroSupport-macos_aarch64/include/_SwiftSyntaxCShims".fixwin))
+            let ldFlags = try XCTUnwrap(target.buildSettings.assignments[.OTHER_LDFLAGS]).flatMap({ $0.values })
+            #if os(Windows)
+            print("ldFlags: ", ldFlags)
+            XCTAssertTrue(ldFlags.contains("\\tmp\\ws\\.build\\prebuilts\\swift-syntax\\\(self.swiftVersion)-MacroSupport-macos_aarch64\\lib/libMacroSupport.a"))
+            #else
+            XCTAssertTrue(ldFlags.contains("/tmp/ws/.build/prebuilts/swift-syntax/\(self.swiftVersion)-MacroSupport-macos_aarch64/lib/libMacroSupport.a"))
+            #endif
+        } else {
+            XCTAssertNil(target.buildSettings.assignments[.OTHER_SWIFT_FLAGS])
+            XCTAssertNil(target.buildSettings.assignments[.OTHER_LDFLAGS])
+        }
+    }
+
+    func testSuccessPath() async throws {
+        let sandbox = AbsolutePath("/tmp/ws")
+        let fs = InMemoryFileSystem()
+
+        let artifact = Data()
+        let (manifest, rootPackage, swiftSyntax) = try initData(artifact: artifact, swiftSyntaxVersion: "600.0.1")
+        let manifestData = try JSONEncoder().encode(manifest)
+
+        let httpClient = HTTPClient { request, progressHandler in
+            guard case .download(let fileSystem, let destination) = request.kind else {
```

Comment:
Suggestion: in order to clarify the test intent, could we add code comment such as 

```
// GIVEN <arrangement>
   // insert code for the test arrangement

// WHEN <action under test>
    // insert code to complete the action

// THEN <expectation 1>
    // insert assertion code

// AND <expectation 2>
    // insert assertion code
```

This comment applies to all net introduces tests